src/Manticoresearch/Connection.php

```diff
@@ -202,13 +202,13 @@ public function getConfig($key =  null)
     }
 
     /**
-     * @param array $params|self
-     * @return array|static
+     * @param array|Connection $params|self
+     * @return Connection
      */
     public static function create($params)
     {
         if (is_array($params)) {
-            return new static($params);
+            return new Connection($params);
```

Comment:
Wouldn't be better to use `self`? If you extend the Connection class, `create` will make `Connection` class instead of your custom one.

Reply:
hi Adrian,

Yes you are correct, I was concentrating on getting the PHPStan checks to pass.  I should stop coding at midnight :)

Cheers

G